src/main/scala/com/persist/JsonFormat.scala

```diff
@@ -203,6 +203,10 @@ package object json {
       def read(json: Json): Set[T] = extractSeq[T](json).toSet
     }
 
+    implicit def setCollection[T: ReadCodec] = new ReadCodec[scala.collection.Set[T]] {
```

Comment:
This is the new feature we're adding? Do we have tests showing how this is used? What happens if I pass null? None?